src/utils/roCrateSingleton.ts

```diff
@@ -0,0 +1,135 @@
+import JSZip from 'jszip';
+import { ROCrate } from 'ro-crate';
+
+const RoCrateSingleton = (() => {
+  let trackProperties = null;
+  let trackPropertiesURL = null;
+  let trackCrate = null;
+  let trackCrateZip = null;
+  let previewHtml = null;
+  let currentCrateUrl = null;
+  let specifiedCrateFolder = null;
+
+  // const determineFilePath = (fileName) => {
+  //   console.log('files', trackCrateZip.files);
+  //   return specifiedCrateFolder &&
+  //     trackCrateZip.files[`${specifiedCrateFolder}/ro-crate-preview.html`]
+  //     ? `${specifiedCrateFolder}/${fileName}`
+  //     : fileName;
+  // };
+
+  const determineFilePath = (fileName) => {
+    return specifiedCrateFolder
+      ? `${specifiedCrateFolder}/${fileName}`
+      : fileName;
+  };
+
+  const extractDetailsFromCrateZip = async (crateUrl) => {
+    currentCrateUrl = crateUrl;
+    try {
+      const response = await fetch(crateUrl);
+      if (response.status === 200 || response.status === 0) {
```

Comment:
Is there a known case for status 0 being success?

Reply:
@SandyRogers I do not know of any such case. I got this from the legacy code. I assumed there was a reason for it.
So far I have not encountered it, and have thus removed that extra condition.